src/Build.UnitTests/BackEnd/BuildManager_Tests.cs

```diff
@@ -4408,5 +4408,29 @@ public void TaskInputLoggingIsExposedToTasks(string taskFactory, bool taskInputL
 
             Assert.Equal(BuildResultCode.Success, result.OverallResult);
         }
+
+        [Fact]
+        public void ProjectWithNoTargets()
+        {
+            string contents = @"<Project />";
+
+            BuildRequestData data = GetBuildRequestData(contents);
+            BuildResult result = _buildManager.Build(_parameters, data);
+            Assert.Equal(BuildResultCode.Failure, result.OverallResult);
+
+            _logger.AssertLogContains("MSB4040");
+        }
+
+        [Fact]
+        public void ProjectWithNoTargetsGraph()
+        {
+            string contents = @"<Project />";
+
+            GraphBuildRequestData data = GetGraphBuildRequestData(contents);
```

Comment:
Not related to this PR - just wondering why BuildRequestData and GraphBuildRequestData doesn't have a common ancestor (interface or abstract class) so that we do not have to have that much code duplication (The `Build` methods are otherwise identical, the calls to `Build` - as e.g. here seems identical as well). Do you know about any strong reasons why they are kept copletely separate? (I'm contemplating creating a refactoring item for that)

Reply:
No particular reason. Might help reduce some of the code duplicate in fact.